internal/pkg/aws/s3/s3.go

```diff
@@ -206,6 +209,78 @@ func (s *S3) isBucketExists(bucket string) (bool, error) {
 	return true, nil
 }
 
+// GetBucketTree retrieves the objects in an S3 bucket and creates an ASCII tree representing their folder structure.
+func (s *S3) GetBucketTree(bucket string) (string, error) {
+	exists, err := s.isBucket(bucket)
+	if err != nil {
+		return "", fmt.Errorf("unable to determine the existence of bucket %s: %w", bucket, err)
```

Comment:
nit: probably don't need to wrap since `isBucket` already wraps it!
```suggestion
		return "", err
```